ecl2df/summary.py

```diff
@@ -219,6 +220,12 @@ def df(
     This is a thin wrapper for EclSum.pandas_frame, by adding
     support for string mnenomics for the time index.
 
+    The dataframe is always indexed by DATE, and the datatype for the
+    index will be usually be datetime64[ns] as long as all dates are
+    before year 2262. If a longer time range is detected, the index.dtype
+    will be object, and consisting of datetime.datetime() objects. The index
+    is always named "DATE".
```

Comment:
Not possible to always return `object` (`datetime.datetime`)? `pandas` living its own life? Not a fan of varying output types... Better to handle it here than in every dependecy?

Reply:
I think we must store it as a string if we want to guarantee `object` all the time, and this is not tempting to enforce. Pandas is very eager about its DatetimeIndex.

Reply:
hmm, ok. Then I'm tempted to ask (though this is quite breaking, so probably not for this PR): Can we not just have `DATE` as a regular column and not care about the index at all? 😅

Reply:
I would prefer to always have `datetime.datetime()` in the index, and to Pandas, that is an index of type `object`. But pandas operations (like `apply()`) on indices with datetime.datetime jump to a DatetimeIndex when you poke it, as long as you are before year 2262. So I _think_ this variable output is something the API user must live with.

Reply:
Ok, I'll resign 😣🏳😅